cabal-install/Distribution/Client/SrcDist.hs

```diff
@@ -179,6 +181,8 @@ allPackageSourceFiles verbosity packageDir = do
           "Exception was: " ++ show e
 
   -- Run setup sdist --list-sources=TMPFILE
-  ret <- doListSources `catchIO` (\e -> onFailedListSources e >> return [])
-  removeExistingFile file
-  return ret
+  r <- doListSources `catchIO` (\e -> onFailedListSources e >> return [])
+  -- Ensure that we've closed the 'readFile' handle before we exit the
+  -- temporary directory.
+  _ <- evaluate (length r)
```

Comment:
I'd put the forcing at the source, which is:

    fmap lines . readFile $ file